.clang-tidy

```diff
@@ -12,13 +12,21 @@ Checks: |
   -cppcoreguidelines-pro-bounds-pointer-arithmetic
   -cppcoreguidelines-init-variables
   -cppcoreguidelines-pro-type-member-init
+  -cppcoreguidelines-avoid-do-while
+  -cppcoreguidelines-pro-type-const-cast
+  -cppcoreguidelines-pro-bounds-array-to-pointer-decay
+  -cppcoreguidelines-avoid-const-or-ref-data-members
+  -cppcoreguidelines-pro-bounds-constant-array-index
   bugprone-*
   -bugprone-sizeof-expression
   -bugprone-narrowing-conversions
   -bugprone-macro-parentheses
+  -bugprone-easily-swappable-parameters
```

Comment:
✅ These seem over the top:
```
/home/runner/work/celeritas/celeritas/src/orange/orangeinp/IntersectRegion.cc:217:20: warning: 2 adjacent parameters of 'Cylinder' of similar type ('celeritas::real_type') are easily swapped by mistake [bugprone-easily-swappable-parameters]
Cylinder::Cylinder(real_type radius, real_type halfheight)
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
Right, this one has quite a [few options](https://clang.llvm.org/extra/clang-tidy/checks/bugprone/easily-swappable-parameters.html), maybe at 5 or 6 parameter it is a bit less over the top